yarn-project/foundation/src/json-rpc/server/json_rpc_server.ts

```diff
@@ -80,13 +80,20 @@ export class JsonRpcServer {
         }
         router.post(`/${method}`, async (ctx: Koa.Context) => {
           const { params = [], jsonrpc, id } = ctx.request.body as any;
-          const result = await this.proxy.call(method, params);
-          ctx.body = {
-            jsonrpc,
-            id,
-            result: convertBigintsInObj(result),
-          };
-          ctx.status = 200;
+          try {
+            const result = await this.proxy.call(method, params);
+            ctx.body = {
+              jsonrpc,
+              id,
+              result: convertBigintsInObj(result),
+            };
+            ctx.status = 200;
+          } catch (err: any) {
+            // Propagate the error message to the client. Plenty of the errors are expected to occur (e.g. adding
+            // a duplicate recipient) so this is necessary.
+            ctx.status = 409;
```

Comment:
I originally though 409 fits better this case because the errors I had in mind were data conflicts (e.g. recipient already registered). But it's true that other kinds of errors could be captured here as well so I changed it in [0b4a3d1](https://github.com/AztecProtocol/aztec-packages/pull/1568/commits/0b4a3d1c23b40a7406b95414b6d58e0446a19da1).